lib/Analysis/MulDepthAnalysis/MulDepthAnalysis.cpp

```diff
@@ -0,0 +1,64 @@
+#include "lib/Analysis/MulDepthAnalysis/MulDepthAnalysis.h"  // from @llvm-project
+
+#include "lib/Dialect/Openfhe/IR/OpenfheOps.h"
+#include "llvm/include/llvm/ADT/TypeSwitch.h"  // from @llvm-project
+#include "llvm/include/llvm/Support/Debug.h"   // from @llvm-project
+#include "mlir/include/mlir/IR/Operation.h"    // from @llvm-project
+#include "mlir/include/mlir/IR/Value.h"        // from @llvm-project
+#include "mlir/include/mlir/IR/Visitors.h"     // from @llvm-project
+#include "mlir/include/mlir/Support/LLVM.h"    // from @llvm-project
+
+#define DEBUG_TYPE "openfhe-muldepth-analysis"
+
+namespace mlir {
+namespace heir {
+namespace mul_depth_analysis {
+
+void MulDepthAnalysis::visitOperation(
+    Operation *op, ArrayRef<const MulDepthLattice *> operands,
+    ArrayRef<MulDepthLattice *> results) {
+  llvm::TypeSwitch<Operation &>(*op)
+      .Case<openfhe::MulOp, openfhe::MulPlainOp, openfhe::MulConstOp,
+            openfhe::MulNoRelinOp>([&](auto mulOp) {
+        LLVM_DEBUG(
+            { llvm::dbgs() << "Visiting Mul: " << mulOp->getName() << "\n"; });
```

Comment:
note, for a single line you can omit the `{ }` and the semicolon in the inner statement. Maybe this is just slightly over the line limit?

```suggestion
        LLVM_DEBUG(llvm::dbgs() << "Visiting Mul: " << mulOp->getName() << "\n");
```